node/src/db/db_handlers.rs

```diff
@@ -0,0 +1,1140 @@
+                    }
+                };
+            }
+            Err(error) => {
+                insert_ancestor_timestamp_tx.rollback().await.unwrap();
+                return Err(DBErrors::TupleNotInserted {
+                    error: error.to_string(),
+                });
+            }
+        };
+    }
+    async fn insert_cohort_id(&self, cohort_id_tuple: CohortIdTuple) -> Result<u64, DBErrors> {
+        let mut insert_cohort_id_tx = self.db_connection.lock().await.begin().await.unwrap();
+        match sqlx::query("INSERT INTO CohortIds (id) VALUES (?)")
+            .bind(cohort_id_tuple.id)
+            .execute(&mut *insert_cohort_id_tx)
+            .await
+        {
+            Ok(query_result) => {
+                match insert_cohort_id_tx.commit().await {
+                    Ok(_) => {
+                        log::info!("Insert cohort id transaction committed");
+                        return Ok(query_result.rows_affected());
+                    }
+                    Err(error) => {
+                        return Err(DBErrors::InsertionTransactionNotCommitted {
+                            error: error.to_string(),
+                            query_name: "Insertion of cohort Id".to_string(),
+                        })
+                    }
+                };
+            }
+            Err(error) => {
+                insert_cohort_id_tx.rollback().await.unwrap();
+                return Err(DBErrors::TupleNotInserted {
+                    error: error.to_string(),
+                });
+            }
+        };
+    }
+    async fn insert_cohort_tuple(&self, cohort_tuple: CohortTuple) -> Result<u64, DBErrors> {
+        let mut insert_cohort_tx = self.db_connection.lock().await.begin().await.unwrap();
+        match sqlx::query("INSERT INTO Cohorts (bead_id, cohort_id) VALUES (?, ?)")
+            .bind(cohort_tuple.bead_id)
+            .bind(cohort_tuple.cohort_id)
+            .execute(&mut *insert_cohort_tx)
+            .await
+        {
+            Ok(query_result) => {
+                match insert_cohort_tx.commit().await {
+                    Ok(_) => {
+                        log::info!("Insert cohort committed");
+                        return Ok(query_result.rows_affected());
+                    }
+                    Err(error) => {
+                        return Err(DBErrors::InsertionTransactionNotCommitted {
+                            error: error.to_string(),
+                            query_name: "Insertion of cohort".to_string(),
+                        })
+                    }
+                };
+            }
+            Err(error) => {
+                insert_cohort_tx.rollback().await.unwrap();
+                return Err(DBErrors::TupleNotInserted {
+                    error: error.to_string(),
+                });
+            }
+        };
+    }
+    async fn insert_relative_tuple(&self, relative_tuple: RelativeTuple) -> Result<u64, DBErrors> {
```

Comment:
Again the only way this should happen is by inserting a bead. We don't want to make the db inconsistent by adding relatives when they're not specified directly in a bead. I think we can delete this function.